### PR TITLE
Updated the edit button for location and fixed the modal functionality

### PR DESCRIPTION
**Pull request by robinpiets**
I've fixed the functionality for the Modal and made sure that the normal EditButton is still available on the OLO flow. Also, the Modal has an extra option to send a component that can be used as openButton.

Please make sure:
- [x] to add a label
- [x] to add one or more reviewers
- [x] you followed our checklist for [functional testing](https://github.com/Amsterdam/vergunningcheck/blob/master/TESTING.md)
- [x] you added the necessary documentation (either in the markdown files or inline)
- [x] ~~you added the necessary automated tests~~ To be added later

### CODE REVIEW

File: apps/client/src/components/Location/ChangeAddressModal.tsx [hunk 1]
```diff
@@ -1,16 +1,28 @@
...
 import Modal from "../Modal";
 
 const ChangeAddressModal: React.FC<{
   matomoTrackEvent: Function;
-}> = ({ matomoTrackEvent }) => {
+  setActiveState: Function;
+  slug: string;
+  sttrFile: boolean;
+}> = ({ matomoTrackEvent, setActiveState, slug, sttrFile }) => {
+  // What's taking the React custom hooks so long?
```
Comment (afjlambert) [inline apps/client/src/components/Location/ChangeAddressModal.tsx]: imtr refactor

File: apps/client/src/components/Modal.tsx [hunk 2]
```diff
@@ -67,17 +69,23 @@ const Modal: React.FC<ModalProps> = ({
     }
   };
 
+  // How to do this properly? @Sven
```
Comment (afjlambert) [inline apps/client/src/components/Modal.tsx]: Did the solution we discussed work for you?

File: apps/client/src/components/Location/ChangeAddressModal.tsx [hunk 3]
```diff
@@ -1,16 +1,28 @@
...
+import { SessionContext, SessionDataType } from "../../context";
 import withTracking from "../../hoc/withTracking";
 import Modal from "../Modal";
 
 const ChangeAddressModal: React.FC<{
   matomoTrackEvent: Function;
-}> = ({ matomoTrackEvent }) => {
+  setActiveState: Function;
+  slug: string;
+  sttrFile: boolean;
```
Comment (afjlambert) [inline apps/client/src/components/Location/ChangeAddressModal.tsx]: It's confusing to have the sttrFile having type boolean here. Either make it a string and do !! or use hasSTTR.

**Review by afjlambert — CHANGES_REQUESTED** [review]

File: apps/client/src/components/Location/ChangeAddressModal.tsx (hunk 3 above)
Comment (robinpiets) [reply apps/client/src/components/Location/ChangeAddressModal.tsx]: Thanks! Replaced this

File: apps/client/src/components/Location/ChangeAddressModal.tsx (hunk 1 above)
Comment (robinpiets) [reply apps/client/src/components/Location/ChangeAddressModal.tsx]: Removed the comment

File: apps/client/src/components/Modal.tsx (hunk 2 above)
Comment (robinpiets) [reply apps/client/src/components/Modal.tsx]: Works perfectly!

File: apps/client/src/components/RegisterLookupSummary.js [hunk 4]
```diff
@@ -82,14 +60,14 @@ const RegisterLookupSummary = ({
       </Paragraph>
       <Paragraph
         data-testid={LOCATION_RESTRICTION_CITYSCAPE}
-        gutterBottom={displayZoningPlans ? 16 : 0}
+        gutterBottom={!hasSTTR ? 16 : 0}
```
Comment (afjlambert) [inline apps/client/src/components/RegisterLookupSummary.js]: You can flip the values after ternary operator and get rid of the `!`.

**Review by afjlambert — APPROVED** [review]

File: apps/client/src/components/RegisterLookupSummary.js (hunk 4 above)
Comment (robinpiets) [reply apps/client/src/components/RegisterLookupSummary.js]: Nice

**Pull request merged**